src/storage/heap_file.c

```diff
@@ -7858,9 +7867,6 @@ heap_next_internal (THREAD_ENTRY * thread_p, const HFID * hfid, OID * class_oid,
       class_oid = &scan_cache->node.class_oid;
     }
 
-  PGBUF_INIT_WATCHER (&curr_page_watcher, PGBUF_ORDERED_HEAP_NORMAL, hfid);
-  PGBUF_INIT_WATCHER (&old_page_watcher, PGBUF_ORDERED_HEAP_NORMAL, hfid);
```

Comment:
I think it's good idea to make use of `scan_cache->page_watcher`instead of `curr_page_watcher`. It removes the redundant copies to and from the local variable. I don't see any side effect from this. :)

However, I can't see why `old_page_watcher` also needs to be in the scan_cache. It seems used locally in this function and it does not reduce the call of `pgbuf_replace_watcher`. Could you tell me the reason?